src/utils/arrayUtils.ts

```diff
@@ -41,3 +41,41 @@ export function removeFalsy<const T>(
 ): T[] {
   return array.filter((e) => e) as T[]
 }
+
+export type RecordKeyType = string | number | symbol
+export function keyArrayBy<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
+  array: T[],
+  selector: K,
+  arrayValue?: true,
+): Record<T[K], T[]>
+export function keyArrayBy<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
+  array: T[],
+  selector: K,
+  arrayValue: false,
+): Record<T[K], T>
+export function keyArrayBy<T extends { [K in keyof T]: RecordKeyType }, K extends keyof T>(
+  array: T[],
+  selector: K,
+  arrayValue = true,
+) {
+  if (arrayValue) {
+    return array.reduce(
```

Comment:
Why do we need else if we `return` on each condition?

Reply:
You are right! but with the refactor it is not needed now